### DISCUSSION TITLE
Installation failed on CentOS 7.9 64bit

### DISCUSSION THREAD

**Discussion by skyshe** [netbox-community/netbox#5669] [Q&A] [answered]
Hi Everybody

I encountered a netbox installation. My system is CentOS 7.9 64bit. According to the netbox documentation, both pg and redis are installed and tested, but an error is reported during the execution of the upgrade script.
How can I solve it?

### Environment
* Python version:  python -V Python 2.7.5 / python3 -V  Python 3.6.8
* NetBox version: netbox-2.10.3

### Steps to Reproduce
1. 
2. 
3. 

### Observed Behavior

**err  info：**
[root@ipam ~]# sudo /opt/netbox/upgrade.sh
Removing old virtual environment...
Creating a new virtual environment at /opt/netbox-2.10.3/venv...
Installing Python system packages (pip3 install wheel)...
Collecting wheel
  Downloading http://mirrors.aliyun.com/pypi/packages/65/63/39d04c74222770ed1589c0eaba06c05891801219272420b40311cd60c880/wheel-0.36.2-py2.py3-none-any.whl
Installing collected packages: wheel
Successfully installed wheel-0.36.2
You are using pip version 9.0.3, however version 20.3.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Installing core dependencies (pip3 install -r requirements.txt)...
Collecting Django==3.1.3 (from -r requirements.txt (line 1))
  Downloading http://mirrors.aliyun.com/pypi/packages/7f/17/16267e782a30ea2ce08a9a452c1db285afb0ff226cfe3753f484d3d65662/Django-3.1.3-py3-none-any.whl (7.8MB)
    100% |████████████████████████████████| 7.8MB 1.3MB/s 
Collecting django-cacheops==5.1 (from -r requirements.txt (line 2))
  Downloading http://mirrors.aliyun.com/pypi/packages/5d/0c/530f4f32e770c1f0df3ecc988f97ef548ae59ce5a49caae8bcad96afcaec/django_cacheops-5.1-py2.py3-none-any.whl
Collecting django-cors-headers==3.5.0 (from -r requirements.txt (line 3))
  Downloading http://mirrors.aliyun.com/pypi/packages/51/99/41898d5eddb68bc5545c08f5315e00d6036cc6f1aba5fd824042dd1b04ea/django_cors_headers-3.5.0-py3-none-any.whl
Collecting django-debug-toolbar==3.1.1 (from -r requirements.txt (line 4))
  Downloading http://mirrors.aliyun.com/pypi/packages/22/14/864c8805134fa25a8165f895c630cdd08d8ae4dd2d3557878624a9f686d2/django_debug_toolbar-3.1.1-py3-none-any.whl (197kB)
    100% |████████████████████████████████| 204kB 3.6MB/s 
Collecting django-filter==2.4.0 (from -r requirements.txt (line 5))
  Downloading http://mirrors.aliyun.com/pypi/packages/71/2b/b2fe483c3095b6222725dd05f9ad9e6ed6cb7347c154fdbd80238d36f1a8/django_filter-2.4.0-py3-none-any.whl (73kB)
    100% |████████████████████████████████| 81kB 4.2MB/s 
Collecting django-mptt==0.11.0 (from -r requirements.txt (line 6))
  Downloading http://mirrors.aliyun.com/pypi/packages/83/86/78a64e1aa72dc6156a1a4f1f830564c932b0900460607def7ccd56a1df80/django_mptt-0.11.0-py2.py3-none-any.whl (109kB)
    100% |████████████████████████████████| 112kB 4.1MB/s 
Collecting django-pglocks==1.0.4 (from -r requirements.txt (line 7))
Collecting django-prometheus==2.1.0 (from -r requirements.txt (line 8))
  Downloading http://mirrors.aliyun.com/pypi/packages/9b/1b/af5e3c82da897618c90836e7ff3c17f6c3f1dfb82ba33cc20815b2a1f8be/django_prometheus-2.1.0-py2.py3-none-any.whl
Collecting django-rq==2.4.0 (from -r requirements.txt (line 9))
  Downloading http://mirrors.aliyun.com/pypi/packages/53/bb/fb8306b3cba7f6cbae57a1f4d17c3abd4aeb767c1febf97cf1d38d3c724f/django_rq-2.4.0-py2.py3-none-any.whl (46kB)
    100% |████████████████████████████████| 51kB 5.6MB/s 
Collecting django-tables2==2.3.3 (from -r requirements.txt (line 10))
  Downloading http://mirrors.aliyun.com/pypi/packages/dd/1d/27bcbfe4dc80c2e38ee80f2d2a35555a00b44731f6155c60a214a4494490/django_tables2-2.3.3-py2.py3-none-any.whl (91kB)
    100% |████████████████████████████████| 92kB 5.4MB/s 
Collecting django-taggit==1.3.0 (from -r requirements.txt (line 11))
  Downloading http://mirrors.aliyun.com/pypi/packages/56/3e/dde2d7151bc0c4ac65d225e611a85e54a897c551507e8eca2c06a083f3f4/django_taggit-1.3.0-py3-none-any.whl (45kB)
    100% |████████████████████████████████| 51kB 10.2MB/s 
Collecting django-timezone-field==4.0 (from -r requirements.txt (line 12))
  Downloading http://mirrors.aliyun.com/pypi/packages/b3/23/ac06d9dcc9450b8ed51a7f861630b5ae4c1779c53401c83703fb87fbef39/django_timezone_field-4.0-py3-none-any.whl
Collecting djangorestframework==3.12.2 (from -r requirements.txt (line 13))
  Downloading http://mirrors.aliyun.com/pypi/packages/8e/42/4cd19938181a912150e55835109b1933be26b776f3d4fb186491968dc41d/djangorestframework-3.12.2-py3-none-any.whl (957kB)
    100% |████████████████████████████████| 962kB 1.3MB/s 
Collecting drf-yasg[validation]==1.20.0 (from -r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/fc/5c/4584321c7a7fd85429f73fd4c3406b09309e3a071e250177739384a4ab48/drf_yasg-1.20.0-py2.py3-none-any.whl (1.6MB)
    100% |████████████████████████████████| 1.6MB 1.3MB/s 
Collecting gunicorn==20.0.4 (from -r requirements.txt (line 15))
  Downloading http://mirrors.aliyun.com/pypi/packages/69/ca/926f7cd3a2014b16870086b2d0fdc84a9e49473c68a8dff8b57f7c156f43/gunicorn-20.0.4-py2.py3-none-any.whl (77kB)
    100% |████████████████████████████████| 81kB 5.5MB/s 
Collecting Jinja2==2.11.2 (from -r requirements.txt (line 16))
  Downloading http://mirrors.aliyun.com/pypi/packages/30/9e/f663a2aa66a09d838042ae1a2c5659828bb9b41ea3a6efa20a20fd92b121/Jinja2-2.11.2-py2.py3-none-any.whl (125kB)
    100% |████████████████████████████████| 133kB 3.9MB/s 
Collecting Markdown==3.3.3 (from -r requirements.txt (line 17))
  Downloading http://mirrors.aliyun.com/pypi/packages/ac/ef/24a91ca96efa0d7802dffb83ccc7a3c677027bea19ec3c9ee80be740408e/Markdown-3.3.3-py3-none-any.whl (96kB)
    100% |████████████████████████████████| 102kB 4.8MB/s 
Collecting netaddr==0.8.0 (from -r requirements.txt (line 18))
  Downloading http://mirrors.aliyun.com/pypi/packages/ff/cd/9cdfea8fc45c56680b798db6a55fa60a22e2d3d3ccf54fc729d083b50ce4/netaddr-0.8.0-py2.py3-none-any.whl (1.9MB)
    100% |████████████████████████████████| 1.9MB 345kB/s 
Collecting Pillow==8.0.1 (from -r requirements.txt (line 19))
  Downloading http://mirrors.aliyun.com/pypi/packages/5f/19/d4c25111d36163698396f93c363114cf1cddbacb24744f6612f25b6aa3d0/Pillow-8.0.1-cp36-cp36m-manylinux1_x86_64.whl (2.2MB)
    100% |████████████████████████████████| 2.2MB 1.2MB/s 
Collecting psycopg2-binary==2.8.6 (from -r requirements.txt (line 20))
  Downloading http://mirrors.aliyun.com/pypi/packages/f2/1b/720b36697158113ca1b2221a8e96a470088ccf3770d182214689d1a96a07/psycopg2_binary-2.8.6-cp36-cp36m-manylinux1_x86_64.whl (3.0MB)
    100% |████████████████████████████████| 3.0MB 1.3MB/s 
Collecting pycryptodome==3.9.9 (from -r requirements.txt (line 21))
  Downloading http://mirrors.aliyun.com/pypi/packages/2b/6f/7e38d7c97fbbc3987539c804282c33f56b6b07381bf2390deead696440c5/pycryptodome-3.9.9-cp36-cp36m-manylinux1_x86_64.whl (13.7MB)
    100% |████████████████████████████████| 13.7MB 1.3MB/s 
Collecting PyYAML==5.3.1 (from -r requirements.txt (line 22))
Collecting svgwrite==1.4 (from -r requirements.txt (line 23))
  Downloading http://mirrors.aliyun.com/pypi/packages/1c/85/1dc25b36c3ac4f3fe285d33065fc0f2ea7bdfb9209d6369e01a3e8ef6252/svgwrite-1.4-py3-none-any.whl (66kB)
    100% |████████████████████████████████| 71kB 5.7MB/s 
Collecting sqlparse>=0.2.2 (from Django==3.1.3->-r requirements.txt (line 1))
  Downloading http://mirrors.aliyun.com/pypi/packages/14/05/6e8eb62ca685b10e34051a80d7ea94b7137369d8c0be5c3b9d9b6e3f5dae/sqlparse-0.4.1-py3-none-any.whl (42kB)
    100% |████████████████████████████████| 51kB 5.1MB/s 
Collecting asgiref<4,>=3.2.10 (from Django==3.1.3->-r requirements.txt (line 1))
  Downloading http://mirrors.aliyun.com/pypi/packages/89/49/5531992efc62f9c6d08a7199dc31176c8c60f7b2548c6ef245f96f29d0d9/asgiref-3.3.1-py3-none-any.whl
Collecting pytz (from Django==3.1.3->-r requirements.txt (line 1))
  Downloading http://mirrors.aliyun.com/pypi/packages/89/06/2c2d3034b4d6bf22f2a4ae546d16925898658a33b4400cfb7e2c1e2871a3/pytz-2020.5-py2.py3-none-any.whl (510kB)
    100% |████████████████████████████████| 512kB 1.2MB/s 
Collecting redis>=3.0.0 (from django-cacheops==5.1->-r requirements.txt (line 2))
  Downloading http://mirrors.aliyun.com/pypi/packages/a7/7c/24fb0511df653cf1a5d938d8f5d19802a88cef255706fdda242ff97e91b7/redis-3.5.3-py2.py3-none-any.whl (72kB)
    100% |████████████████████████████████| 81kB 5.5MB/s 
Collecting six>=1.4.0 (from django-cacheops==5.1->-r requirements.txt (line 2))
  Downloading http://mirrors.aliyun.com/pypi/packages/ee/ff/48bde5c0f013094d729fe4b0316ba2a24774b3ff1c52d924a8a4cb04078a/six-1.15.0-py2.py3-none-any.whl
Collecting funcy<2.0,>=1.8 (from django-cacheops==5.1->-r requirements.txt (line 2))
  Downloading http://mirrors.aliyun.com/pypi/packages/66/89/479de0afbbfb98d1c4b887936808764627300208bb771fcd823403645a36/funcy-1.15-py2.py3-none-any.whl
Collecting django-js-asset (from django-mptt==0.11.0->-r requirements.txt (line 6))
  Downloading http://mirrors.aliyun.com/pypi/packages/aa/2d/98089cf51c8e83bc70723021390b94a3638a4a0ce30a47e2e70476b2095d/django_js_asset-1.2.2-py2.py3-none-any.whl
Collecting prometheus-client>=0.7 (from django-prometheus==2.1.0->-r requirements.txt (line 8))
  Downloading http://mirrors.aliyun.com/pypi/packages/f4/7e/ef341c67ed43ad2e39633a35c28b77bc555f9572f4df4fee11c2b467db00/prometheus_client-0.9.0-py2.py3-none-any.whl (53kB)
    100% |████████████████████████████████| 61kB 4.9MB/s 
Collecting rq>=1.2 (from django-rq==2.4.0->-r requirements.txt (line 9))
  Downloading http://mirrors.aliyun.com/pypi/packages/2e/47/a04457e5ff635d85af9dba41255ba3b23bf97c894c38967ceae0f7652611/rq-1.7.0-py2.py3-none-any.whl (62kB)
    100% |████████████████████████████████| 71kB 6.6MB/s 
Collecting ruamel.yaml>=0.15.34 (from drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/7e/39/186f14f3836ac5d2a6a042c8de69988770e8b9abb537610edc429e4914aa/ruamel.yaml-0.16.12-py2.py3-none-any.whl (111kB)
    100% |████████████████████████████████| 112kB 4.1MB/s 
Collecting uritemplate>=3.0.0 (from drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/bf/0c/60d82c077998feb631608dca3cc1fe19ac074e772bf0c24cf409b977b815/uritemplate-3.0.1-py2.py3-none-any.whl
Collecting coreapi>=2.3.3 (from drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/fc/3a/9dedaad22962770edd334222f2b3c3e7ad5e1c8cab1d6a7992c30329e2e5/coreapi-2.3.3-py2.py3-none-any.whl
Collecting packaging (from drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/b1/a7/588bfa063e7763247ab6f7e1d994e331b85e0e7d09f853c59a6eb9696974/packaging-20.8-py2.py3-none-any.whl
Collecting coreschema>=0.0.4 (from drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
Collecting inflection>=0.3.1 (from drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/59/91/aa6bde563e0085a02a435aa99b49ef75b0a4b062635e606dab23ce18d720/inflection-0.5.1-py2.py3-none-any.whl
Collecting swagger-spec-validator>=2.1.0; extra == "validation" (from drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/09/de/e78cefbf5838b434b63a789264b79821cb2267f1498fbed23ef8590133e4/swagger_spec_validator-2.7.3-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=3.0 in ./venv/lib/python3.6/site-packages (from gunicorn==20.0.4->-r requirements.txt (line 15))
Collecting MarkupSafe>=0.23 (from Jinja2==2.11.2->-r requirements.txt (line 16))
  Downloading http://mirrors.aliyun.com/pypi/packages/b2/5f/23e0023be6bb885d00ffbefad2942bc51a620328ee910f64abe5a8d18dd1/MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting importlib-metadata; python_version < "3.8" (from Markdown==3.3.3->-r requirements.txt (line 17))
  Downloading http://mirrors.aliyun.com/pypi/packages/f3/ed/da40116a204abb5c4dd1d929346d33e0d29cedb2cedd18ea98f0385dcd92/importlib_metadata-3.4.0-py3-none-any.whl
Collecting importlib-resources; python_version < "3.7" (from netaddr==0.8.0->-r requirements.txt (line 18))
  Downloading http://mirrors.aliyun.com/pypi/packages/82/70/7bf5f275a738629a7252c30c8461502d3658a75363db9f4f88ddbeb9eeac/importlib_resources-5.1.0-py3-none-any.whl
Collecting click>=5.0.0 (from rq>=1.2->django-rq==2.4.0->-r requirements.txt (line 9))
  Downloading http://mirrors.aliyun.com/pypi/packages/d2/3d/fa76db83bf75c4f8d338c2fd15c8d33fdd7ad23a9b5e57eb6c5de26b430e/click-7.1.2-py2.py3-none-any.whl (82kB)
    100% |████████████████████████████████| 92kB 6.3MB/s 
Collecting ruamel.yaml.clib>=0.1.2; platform_python_implementation == "CPython" and python_version < "3.9" (from ruamel.yaml>=0.15.34->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/88/ff/ec25dc01ef04232a9e68ff18492e37dfa01f1f58172e702ad4f38536d41b/ruamel.yaml.clib-0.2.2-cp36-cp36m-manylinux1_x86_64.whl (549kB)
    100% |████████████████████████████████| 552kB 1.3MB/s 
Collecting requests (from coreapi>=2.3.3->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/29/c1/24814557f1d22c56d50280771a17307e6bf87b70727d975fd6b2ce6b014a/requests-2.25.1-py2.py3-none-any.whl (61kB)
    100% |████████████████████████████████| 61kB 3.5MB/s 
Collecting itypes (from coreapi>=2.3.3->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/3f/bb/3bd99c7cd34d4a123b2903e16da364f6d2078b1c3a3530a8ad105c668104/itypes-1.2.0-py2.py3-none-any.whl
Collecting pyparsing>=2.0.2 (from packaging->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/8a/bb/488841f56197b13700afd5658fc279a2025a39e22449b7cf29864669b15d/pyparsing-2.4.7-py2.py3-none-any.whl (67kB)
    100% |████████████████████████████████| 71kB 4.2MB/s 
Collecting jsonschema (from swagger-spec-validator>=2.1.0; extra == "validation"->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/c5/8f/51e89ce52a085483359217bc72cdbf6e75ee595d5b1d4b5ade40c7e018b8/jsonschema-3.2.0-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 4.8MB/s 
Collecting typing-extensions>=3.6.4; python_version < "3.8" (from importlib-metadata; python_version < "3.8"->Markdown==3.3.3->-r requirements.txt (line 17))
  Downloading http://mirrors.aliyun.com/pypi/packages/60/7a/e881b5abb54db0e6e671ab088d079c57ce54e8a01a3ca443f561ccadb37e/typing_extensions-3.7.4.3-py3-none-any.whl
Collecting zipp>=0.5 (from importlib-metadata; python_version < "3.8"->Markdown==3.3.3->-r requirements.txt (line 17))
  Downloading http://mirrors.aliyun.com/pypi/packages/41/ad/6a4f1a124b325618a7fb758b885b68ff7b058eec47d9220a12ab38d90b1f/zipp-3.4.0-py3-none-any.whl
Collecting idna<3,>=2.5 (from requests->coreapi>=2.3.3->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/a2/38/928ddce2273eaa564f6f50de919327bf3a00f091b5baba8dfa9460f3a8a8/idna-2.10-py2.py3-none-any.whl (58kB)
    100% |████████████████████████████████| 61kB 4.8MB/s 
Collecting urllib3<1.27,>=1.21.1 (from requests->coreapi>=2.3.3->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/f5/71/45d36a8df68f3ebb098d6861b2c017f3d094538c0fb98fa61d4dc43e69b9/urllib3-1.26.2-py2.py3-none-any.whl (136kB)
    100% |████████████████████████████████| 143kB 8.8MB/s 
Collecting certifi>=2017.4.17 (from requests->coreapi>=2.3.3->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/5e/a0/5f06e1e1d463903cf0c0eebeb751791119ed7a4b3737fdc9a77f1cdfb51f/certifi-2020.12.5-py2.py3-none-any.whl (147kB)
    100% |████████████████████████████████| 153kB 6.2MB/s 
Collecting chardet<5,>=3.0.2 (from requests->coreapi>=2.3.3->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/19/c7/fa589626997dd07bd87d9269342ccb74b1720384a4d739a1872bd84fbe68/chardet-4.0.0-py2.py3-none-any.whl (178kB)
    100% |████████████████████████████████| 184kB 396kB/s 
Collecting attrs>=17.4.0 (from jsonschema->swagger-spec-validator>=2.1.0; extra == "validation"->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
  Downloading http://mirrors.aliyun.com/pypi/packages/c3/aa/cb45262569fcc047bf070b5de61813724d6726db83259222cd7b4c79821a/attrs-20.3.0-py2.py3-none-any.whl (49kB)
    100% |████████████████████████████████| 51kB 9.9MB/s 
Collecting pyrsistent>=0.14.0 (from jsonschema->swagger-spec-validator>=2.1.0; extra == "validation"->drf-yasg[validation]==1.20.0->-r requirements.txt (line 14))
Installing collected packages: sqlparse, asgiref, pytz, Django, redis, six, funcy, django-cacheops, django-cors-headers, django-debug-toolbar, django-filter, django-js-asset, django-mptt, django-pglocks, prometheus-client, django-prometheus, click, rq, django-rq, django-tables2, django-taggit, django-timezone-field, djangorestframework, ruamel.yaml.clib, ruamel.yaml, uritemplate, MarkupSafe, Jinja2, coreschema, idna, urllib3, certifi, chardet, requests, itypes, coreapi, pyparsing, packaging, inflection, typing-extensions, zipp, importlib-metadata, attrs, pyrsistent, jsonschema, PyYAML, swagger-spec-validator, drf-yasg, gunicorn, Markdown, importlib-resources, netaddr, Pillow, psycopg2-binary, pycryptodome, svgwrite
Successfully installed Django-3.1.3 Jinja2-2.11.2 Markdown-3.3.3 MarkupSafe-1.1.1 Pillow-8.0.1 PyYAML-5.3.1 asgiref-3.3.1 attrs-20.3.0 certifi-2020.12.5 chardet-4.0.0 click-7.1.2 coreapi-2.3.3 coreschema-0.0.4 django-cacheops-5.1 django-cors-headers-3.5.0 django-debug-toolbar-3.1.1 django-filter-2.4.0 django-js-asset-1.2.2 django-mptt-0.11.0 django-pglocks-1.0.4 django-prometheus-2.1.0 django-rq-2.4.0 django-tables2-2.3.3 django-taggit-1.3.0 django-timezone-field-4.0 djangorestframework-3.12.2 drf-yasg-1.20.0 funcy-1.15 gunicorn-20.0.4 idna-2.10 importlib-metadata-3.4.0 importlib-resources-5.1.0 inflection-0.5.1 itypes-1.2.0 jsonschema-3.2.0 netaddr-0.8.0 packaging-20.8 prometheus-client-0.9.0 psycopg2-binary-2.8.6 pycryptodome-3.9.9 pyparsing-2.4.7 pyrsistent-0.17.3 pytz-2020.5 redis-3.5.3 requests-2.25.1 rq-1.7.0 ruamel.yaml-0.16.12 ruamel.yaml.clib-0.2.2 six-1.15.0 sqlparse-0.4.1 svgwrite-1.4 swagger-spec-validator-2.7.3 typing-extensions-3.7.4.3 uritemplate-3.0.1 urllib3-1.26.2 zipp-3.4.0
You are using pip version 9.0.3, however version 20.3.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Skipping local dependencies (local_requirements.txt is empty)
Applying database migrations (python3 netbox/manage.py migrate)...
Operations to perform:
  Apply all migrations: admin, auth, circuits, contenttypes, dcim, extras, ipam, secrets, sessions, taggit, tenancy, users, virtualization
Running migrations:
  Applying tenancy.0002_tenant_group_optional...Traceback (most recent call last):
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.SyntaxError: syntax error at or near "WITH ORDINALITY"
LINE 6:                     FROM unnest(c.conkey) WITH ORDINALITY co...
                                                  ^

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "netbox/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/core/management/base.py", line 371, in execute
    output = self.handle(*args, **options)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/core/management/base.py", line 85, in wrapped
    res = handle_func(*args, **kwargs)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/core/management/commands/migrate.py", line 245, in handle
    fake_initial=fake_initial,
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/migrations/operations/fields.py", line 236, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/base/schema.py", line 572, in alter_field
    old_db_params, new_db_params, strict)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/postgresql/schema.py", line 168, in _alter_field
    new_db_params, strict,
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/base/schema.py", line 584, in _alter_field
    fk_names = self._constraint_names(model, [old_field.column], foreign_key=True)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/base/schema.py", line 1188, in _constraint_names
    constraints = self.connection.introspection.get_constraints(cursor, model._meta.db_table)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/postgresql/introspection.py", line 167, in get_constraints
    """, [table_name])
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/cacheops/transaction.py", line 93, in execute
    result = self._no_monkey.execute(self, sql, params)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/opt/netbox-2.10.3/venv/lib64/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: syntax error at or near "WITH ORDINALITY"
LINE 6:                     FROM unnest(c.conkey) WITH ORDINALITY co...
                                                  ^

**Comment by drmsoffall** [comment, marked as answer]
Did you check to make sure the Postgres (and Redis) versions installed are at least the minimum required versions?  WITH ORDINALITY is a Postgres 9.4 feature, and NetBox requires 9.6 or later (per the install docs), so yours may be too old.

**Reply by skyshe** [reply]
Hi

Thank you very much for your guidance. The default version of pg and Redis installed by yum is indeed lower than the requirements. After upgrading pg and Redis according to the ideas you provided, it will be normal after executing the upgrade script. I don’t know how to program, so I can’t see WITH ORDINALITY. The new version of pg is required, so thank you very much for your help.
